### PR TITLE
Refactor: Cleanup NoopTask

### PR DESCRIPTION
**Pull request by kfaraz**
### Changes

- Simplify static `create` methods for `NoopTask`
- Remove usage of `FirehoseFactory` and `IsReadyResult` from `NoopTask` as it was not being used anywhere
- Update tests

<hr>

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java [hunk 1]
```diff
@@ -180,33 +155,18 @@ public int getPriority()
...
-  {
-    final Map<String, Object> context = new HashMap<>();
-    context.put(Tasks.PRIORITY_KEY, priority);
-    return new NoopTask(null, null, null, 0, 0, null, null, context);
+    return new NoopTask(null, null, datasource, 0, 0, null, null);
   }
 
-  @VisibleForTesting
-  public static NoopTask create(String id, int priority)
+  public static NoopTask withPriority(int priority)
```
Comment (AmatyaAvadhanula) [inline indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java:166]: withPriority seems a bit misleading as one may think that the other fields' values would be retained

File: indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java [hunk 2]
```diff
@@ -159,17 +143,8 @@ public void stopGracefully(TaskConfig taskConfig)
   @Override
   public TaskStatus runTask(TaskToolbox toolbox) throws Exception
   {
-    if (firehoseFactory != null) {
-      log.info("Connecting firehose");
-    }
-    try (Firehose firehose = firehoseFactory != null ? firehoseFactory.connect(null, null) : null) {
-
-      log.info("Running noop task[%s]", getId());
-      log.info("Sleeping for %,d millis.", runTime);
```
Comment (AmatyaAvadhanula) [inline indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java:168]: Could this logging be useful for understanding test runs better?

File: indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java [hunk 3]
```diff
@@ -67,18 +62,14 @@ enum IsReadyResult
...
-
   @JsonCreator
   public NoopTask(
       @JsonProperty("id") String id,
       @JsonProperty("groupId") String groupId,
       @JsonProperty("dataSource") String dataSource,
-      @JsonProperty("runTime") long runTime,
+      @JsonProperty("runTime") long runTimeMillis,
       @JsonProperty("isReadyTime") long isReadyTime,
       @JsonProperty("isReadyResult") String isReadyResult,
```
Comment (AmatyaAvadhanula) [inline indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java:72]: Is a non-null value being passed for `isReadyResult`?

File: indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java (hunk 1 above)
Comment (kfaraz) [reply indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java:166]: We could rename it to `ofPriority`.

File: indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java (hunk 2 above)
Comment (kfaraz) [reply indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java:168]: No, I don't see it adding much value. These logs fill up the logs and make it difficult to find important things such as warnings and errors.

File: indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java (hunk 3 above)
Comment (kfaraz) [reply indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java:72]: Not really. I guess we should remove this too.

**Review by AmatyaAvadhanula — APPROVED** [review]
Thank you for the changes, @kfaraz! +1 after all the checks pass

**Pull request merged**